include/libtorrent/bdecode.hpp

```diff
@@ -190,25 +193,35 @@ struct bdecode_token
 			, "we need to assert t >= 0 here");
 	}
 
-	bdecode_token(std::ptrdiff_t off, std::uint32_t next
-		, bdecode_token::type_t t, std::uint8_t header_size = 0)
+	bdecode_token(std::ptrdiff_t const off, std::uint32_t const next
+		, bdecode_token::type_t const t, std::uint32_t const header_size = 0)
 		: offset(std::uint32_t(off))
 		, type(t)
 		, next_item(next)
-		, header(type == string ? std::uint32_t(header_size - 2) : 0)
+		, header(type == string ? std::uint32_t(header_size - 2)
+			: type == long_string ? std::uint32_t(header_size - 8 - 2) : 0)
 	{
-		TORRENT_ASSERT(type != string || header_size >= 2);
+		TORRENT_ASSERT((type != string && type != long_string) || header_size >= 2);
 		TORRENT_ASSERT(off >= 0);
 		TORRENT_ASSERT(off <= max_offset);
 		TORRENT_ASSERT(next <= max_next_item);
 		// the string has 2 implied header bytes, to allow for longer prefixes
-		TORRENT_ASSERT(header_size < 8 || (type == string && header_size < 10));
+		TORRENT_ASSERT(header_size < 8
+			|| (type == string && header_size < 10)
+			|| (type == long_string && header_size < 18));
 		TORRENT_ASSERT(t <= end);
 		static_assert(std::is_unsigned<std::underlying_type<bdecode_token::type_t>::type>::value
 			, "we need to assert t >= 0 here");
 	}
 
-	int start_offset() const { TORRENT_ASSERT(type == string); return int(header) + 2; }
+	int start_offset() const
+	{
+		TORRENT_ASSERT(type == string || type == long_string);
+		if (type == string)
+			return int(header) + 2;
+		else
+			return int(header) + 8 + 2;
```

Comment:
```suggestion
			return int(header) + 10;
```

Reply:
the 8 and the 2 are there for different reasons. I don't want to obfuscate how this offset is computed

Reply:
> the 8 and the 2 are there for different reasons. I don't want to obfuscate how this offset is computed

They have little meaning for reader without some explanation. But having explanation they can be easily "obfuscated".
Otherwise you could use some named constants to make code self descriptive.